### DISCUSSION TITLE
ELK layout options

### DISCUSSION THREAD

**Discussion by SnachBarr** [terrastruct/d2#1981] [Q&A]
If I run `d2 layout elk` I can see I have a few flag options for altering how the ELK engine runs. 

> --elk-algorithm string
> --elk-nodeNodeBetweenLayers int
> --elk-padding string
> --elk-edgeNodeBetweenLayers int
> --elk-nodeSelfLoop int

Am I limited to these 5 available flags? 
It says flags correspond to the ones found at [https://eclipse.dev/elk/reference.html](https://eclipse.dev/elk/reference.html) but there are so many options there that I haven't been able to figure out how to use.

**Comment by cobnett3** [comment]
Ref: terrastruct/d2/issues/1221

**Comment by stereobooster** [comment]
https://github.com/terrastruct/d2/issues/2236
